File: tradflex_back/tradflex/src/main/java/fr/tradflex/model/sousTitre/SousTitre.java
package fr.tradflex.model.sousTitre;

public record SousTitre(int id, String body, int timeCodeBegin, int timeCodeEnd) {
    public String formateToSRT()
    {
        String sousTitreFormatSRT = "";
        sousTitreFormatSRT +=   this.id() + "\n" +
                this.timeCodeBegin() + "-->" + this.timeCodeEnd() + "\n" +
                this.body() + "\n";
        return sousTitreFormatSRT;
    }

    private String formateBodyToSRT()
    {
        String bodyFormateToSRT = "";

        return bodyFormateToSRT;
    }

    public String formateTimeCodeToSRT()
    {
        String timeCodeFormateToSRT = "";

        return timeCodeFormateToSRT;
    }
}

File: tradflex_front/tradflex-app/src/app/sousTitre.ts
import { Projet } from "./projet";

export class SousTitre {
    id: number;
    body: string;
    timeCodeBegin: number;
    timeCodeEnd: number;
    idProject: number;
    constructor(
        id: number,
        body: string,
        timeCodeBegin: number,
        timeCodeEnd: number,
        idProject: number
    ) {
        this.id = id;
        this.body = body;
        this.timeCodeBegin = timeCodeBegin;
        this.timeCodeEnd = timeCodeEnd;
        this.idProject = idProject;
    }
    
    }
    